engine/eConstr.ml

```diff
@@ -570,16 +570,16 @@ let test_constr_universes env sigma leq m n =
           Constr.compare_head_gen_leq_with kind kind leq_universes leq_sorts
             eq_constr' leq_constr' nargs m n
         and leq_constr' nargs m n = m == n || compare_leq nargs m n in
-        compare_leq 0 m n
+        compare_leq nargs m n
```

Comment:
The change is that we thread `nargs` down to here.

Reply:
M'kay. We should do something with these equality functions btw, they're really unreadable.